core/templates/dev/head/domain/exploration/RuleObjectFactory.js

```diff
@@ -0,0 +1,42 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of Rule
+ * domain objects.
+ */
+
+oppia.factory('RuleObjectFactory', [function() {
+  var Rule = function(ruleType, inputs) {
+    this.ruleType = ruleType;
+    this.inputs = inputs;
+  };
+
+  Rule.prototype.toBackendDict = function() {
+    return {
+      rule_type: this.ruleType,
+      inputs: this.inputs
+    };
+  };
+
+  Rule.createNew = function(ruleType, inputs) {
+    return new Rule(ruleType, inputs);
+  };
+
+  Rule.createFromDict = function(ruleDict) {
```

Comment:
Should this be createFromBackendDict (for parity with everything else)? Any calls originating from the frontend should probably use createNew().